browser/browser_context_keyed_service_factories.cc

```diff
@@ -56,6 +61,10 @@ void EnsureBrowserContextKeyedServiceFactoriesBuilt() {
   ntp_background_images::NTPBackgroundImagesBridgeFactory::GetInstance();
 #endif
 
+#if BUILDFLAG(ENABLE_PLAYLIST)
+  playlist::PlaylistEventRouter::GetFactoryInstance();
```

Comment:
@bridiver You recommended this pattern at https://github.com/brave/brave-core/pull/5733#discussion_r458330569
Previously, this router was owned by PlaylistService in browser layer.
As PlaylistService went to component layer, there was no good candidate for owning this router.
This router need to observe PlaylistService.
So, this lazy initialization is introduced inevitably because we don't assume when PlaylistService is initialized.